internal/gatewayapi/route.go

```diff
@@ -173,6 +173,15 @@ func (t *Translator) processHTTPRouteRules(httpRoute *HTTPRouteContext, parentRe
 		dstAddrTypeMap := make(map[ir.DestinationAddressType]int)
 
 		for _, backendRef := range rule.BackendRefs {
+			if len(backendRef.Filters) > 0 {
```

Comment:
should we `continue`, we have been failing fast and early in other places

Reply:
we could, let's fail fast then on this

Reply:
rethinking again, this code should live inside [validateBackendRef](https://github.com/envoyproxy/gateway/blob/ac0243749b01d1d60b32bae88cb0effb71b61102/internal/provider/kubernetes/helpers.go#L215) which gets called here
https://github.com/envoyproxy/gateway/blob/ac0243749b01d1d60b32bae88cb0effb71b61102/internal/gatewayapi/route.go#L1080

which automatically converts the `weight` into a `invalidWeight` ensuring that a percentage of traffic gets 503d

Reply:
looks good, but the field Filters is within gwapiv1.HTTPBackendRef or gwapiv1a1.GRPCBackendRef.
we process BackendRef based on gwapiv1.BackendRef

Reply:
we've had to do something similar for xRoutes, so we created `RouteContext`
https://github.com/envoyproxy/gateway/blob/cc88e9ccd486d14af48138f733ad70ee61a581c6/internal/gatewayapi/contexts.go#L155

may need something similar here

Reply:
if we want to extend it, then sure
thanks for explaining